packages/server-bridge/src/factories/callHandlersSeriallyFactory.js

```diff
@@ -0,0 +1,22 @@
+import Promise from 'bluebird';
+import size from 'lodash/size';
+
+function doCallHandlers(log, handlers, index, { session, socket }) {
+  if (size(handlers) === index) {
+    return session;
+  }
+
+  return Promise
+        .try(() => handlers[index]({ log, session, socket }))
+        .then(nextSession => doCallHandlers(log, handlers, index + 1, { session: nextSession, socket }))
+        .catch((e) => {
+          log.error(e, 'error calling handlers');
+          throw e;
+        });
+}
+
+export default function callHandlersSeriallyFactory({ log }) {
```

Comment:
not sure about calling these factories tbh. they are simply configuring functions. i try to reserve the term factory for when you are building a fully working system. Though obviously an instance of the factory pattern (a la spring or angular)

Reply:
im conflicted